### PR TITLE
Feature tags

### PR DESCRIPTION
**Pull request by acrellin**

### CODE REVIEW

**Comment by acrellin** [comment]
@stefanv @bnaul this is working now. I haven't thought of a great layout yet, though:

image

Thoughts?

File: public/scripts/reducers.js [hunk 1]
```diff
@@ -25,14 +26,59 @@ function datasets(state = [], action) {
...
+      return state;
+  }
+}
+
+
+function features(state={}, action) {
+  switch (action.type) {
+    case Action.RECEIVE_FEATURES: {
+      let tagList = [];
+      for (const feat in action.payload.tags) {
```
Comment (stefanv) [inline public/scripts/reducers.js]: Can you explain what this section does?

File: public/scripts/reducers.js [hunk 2]
```diff
@@ -25,14 +26,59 @@ function datasets(state = [], action) {
...
+          allFeatsList = allFeatsList.concat(action.payload.features_by_category[ctgy]);
+        }
+      }
+      return { ...action.payload,
+               allFeatsList,
+               tagList,
+               checkedTags: tagList.slice(0),
+               featsWithCheckedTags: allFeatsList.slice(0) };
+    }
+    case 'redux-form/CHANGE': {
```
Comment (stefanv) [inline public/scripts/reducers.js]: Why is this part required?

**Comment by stefanv** [comment]
This is all looking very good, Ari.  Just some questions to help me sync up.

**Comment by acrellin** [comment]
Sure @stefanv, I'm available to chat

File: public/scripts/reducers.js (hunk 1 above)
Comment (acrellin) [reply public/scripts/reducers.js]: This section just creates a duplicate-free list of all tags

File: public/scripts/reducers.js (hunk 2 above)
Comment (acrellin) [reply public/scripts/reducers.js]: This is where we catch changes to the selected tags. If we were to have bound a callback to the `onChange` for each of the tag checkbox elements, it would've overridden the default `redux-form` action and the form state wouldn't have updated correctly.

File: public/scripts/Features.jsx [hunk 3]
```diff
@@ -162,7 +181,8 @@ const mapStateToProps = (state, ownProps) => {
 
 const ffMapDispatchToProps = dispatch => (
   {
-    groupToggleCheckedFeatures: prefix => dispatch(Action.groupToggleCheckedFeatures(prefix))
+    groupToggleCheckedFeatures: ctgy => dispatch(Action.groupToggleCheckedFeatures(ctgy)),
```
Comment (stefanv) [inline public/scripts/Features.jsx]: Can we `props.dispatch` directly, or do we need to work through DispatchToProps?

File: public/scripts/reducers.js (hunk 1 above)
Comment (stefanv) [reply public/scripts/reducers.js]: https://twitter.com/alex_gibson/status/554745380005765120

**Review by stefanv — COMMENTED** [review]
This last change made things much, much clearer!

File: public/scripts/Features.jsx (hunk 3 above)
Comment (acrellin) [reply public/scripts/Features.jsx]: hey, that works! I'll remove all the `mapDispatchToProps` stuff from this module for now, and create an issue to do away with it elsewhere, too.

Comment (acrellin) [reply public/scripts/Features.jsx]: Actually there are many places where it is necessary, including elsewhere in this file. I've removed this one, though.

**Comment by acrellin** [comment]
@stefanv I've implemented your suggested changes. Anything else before we merge this?
